### PR TITLE
refactor: simplify close channel logic

### PR DESCRIPTION
**Pull request by snario**

### CODE REVIEW

**Comment by tomclose** [comment]
Could you give some context around this change and why it is required?

File: packages/server-wallet/src/protocols/close-channel.ts [hunk 1]
```diff
@@ -11,28 +11,17 @@ const stageGuard = (guardStage: Stage) => (s: State | undefined): s is State =>
...
-        turnNum: ps.app.latest.turnNum,
-      })));
+  !!ps.app.supported &&
+  !isFinal(ps.app.latestSignedByMe) &&
+  signState({
+    channelId: ps.app.channelId,
+    ...ps.app.supported,
+    turnNum: ps.app.supported.turnNum + (isFinal(ps.app.supported) ? 0 : 1),
+    isFinal: true,
+  });
```
Comment (snario) [inline packages/server-wallet/src/protocols/close-channel.ts:21]: This allows both participants to sign the same states if they have decided to close a channel (one or both has created / sent a CloseChannel objective). It's written more concisely.

File: packages/server-wallet/src/protocols/close-channel.ts [hunk 2]
```diff
@@ -11,28 +11,17 @@ const stageGuard = (guardStage: Stage) => (s: State | undefined): s is State =>
...
 
 const completeCloseChannel = (ps: ProtocolState): CompleteObjective | false =>
-  isFinal(ps.app.supported) &&
+  (ps.app.support || []).every(isFinal) &&
```
Comment (snario) [inline packages/server-wallet/src/protocols/close-channel.ts:24]: This is actually a bug fix I think — you should not consider a channel closed unless every state in the support is final.

Comment (andrewgordstewart) [reply packages/server-wallet/src/protocols/close-channel.ts:24]: I remember making a similar change a couple of months ago. Is there now a test that detects this regression?

**Review by tomclose — APPROVED** [review]

**Pull request merged**